support/src/indirect_intrusive_heap.h

```diff
@@ -33,8 +33,10 @@ namespace crimson {
    *
    * heap_info is a data member pointer as to where the heap data in T
    * is stored.
+   *
+   * K is the branching factor of the heap, default is 2 (binary heap).
    */
-  template<typename I, typename T, IndIntruHeapData T::*heap_info, typename C>
+  template<typename I, typename T, IndIntruHeapData T::*heap_info, typename C, uint K = 2>
```

Comment:
I'd like to think about whether having K be a template argumetn is the best way. It could just as easily be determined at runtime, which then lends itself to being set in the configuration file. What are the arguments for leaving it as a template argument?